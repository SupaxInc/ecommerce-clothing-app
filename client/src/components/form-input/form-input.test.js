import React from "react";
import { shallow } from "enzyme";

import * as redux from 'react-redux';
import FormInput from "./form-input";

describe('FormInput component', () => {
    let wrapper;

    // Mock the properties
    const handleChange = jest.fn();
    const label = "Email";
    const mockProps = {
        value: 'foo@gmail.com',
        name: 'email',
        type: 'email'
    };
    
    beforeEach(() => {
        wrapper = shallow(<FormInput handleChange={handleChange} label={label} {...mockProps}/>);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render the FormInput component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger handleChange when onChange is simulated', () => {
        wrapper.find('.form-input').simulate('change', {
            target: {
                name: 'email',
                value: 'test123@gmail.com'
            }
        });

        expect(handleChange).toHaveBeenCalledWith({
            target: {
                name: 'email',
                value: 'test123@gmail.com'
            }
        });
    });

    it('should render shrink animation because label is not empty', () => {
        expect(wrapper.find('.shrink').length).toBe(1);
    });

    it('should not render shrink animation because label is empty', () => {
        // Mock the properties
        const handleChange = jest.fn();
        const mockProps = {
            value: 'foo@gmail.com',
            name: 'email',
            type: 'email'
        };

        let newWrapper = shallow(<FormInput handleChange={handleChange} label="" {...mockProps}/>);
        expect(newWrapper.find('label').length).toBe(0);
        expect(newWrapper.find('.shrink').length).toBe(0);
    });
});