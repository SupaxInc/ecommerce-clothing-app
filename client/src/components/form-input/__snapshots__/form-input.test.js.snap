// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render the FormInput component 1`] = `ShallowWrapper {}`;
